test/integration-serial/resource_discovery_test.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+require 'test_helper'
+
+class ResourceDiscoveryTest < KubernetesDeploy::IntegrationTest
+  def test_non_prunable_crd
```

Comment:
I'd recommend merging these two tests into a single test that: 
- deploys CRDs of both kinds and asserts that all is well
- deploys the rest of the fixture set (CRs of all kinds + some non-CR) and asserts that results look good
- deploys the non-CRs and asserts that the correct CRs were pruned/left alone

Technically that could easily cover the existing CRD success test as well. Since these tests are slow and can be flakey (inherently, because they hit a live minikube), I try to keep tests that are 100% overlapped by others to a minimum. I think we should move those CRD tests to this file regardless.